index.js

```diff
@@ -1,45 +1,57 @@
 'use strict';
 
 const fs = require('fs');
-const promisify = require('util.promisify');
 
-const promisiedFsRealpath = promisify(fs.realpath);
+function callWithFallbacks(funcName, filepath) {
+  let fallbackToDefault = false;
 
-function realpath(filepath) {
-  if (typeof fs.realpath.native === 'function') {
-    return promisify(fs.realpath.native)(filepath);
+  try {
+    if (typeof fs[funcName].native === 'function') {
+      return fs[funcName].native(filepath);
```

Comment:
how does this work? `fs.realpath.native` takes a callback function

```sh-session
$ node -p "require('fs').realpath.native(process.cwd())"
fs.js:137
    throw new ERR_INVALID_CALLBACK();
    ^

TypeError [ERR_INVALID_CALLBACK]: Callback must be a function
    at makeCallback (fs.js:137:11)
    at Function.realpath.native (fs.js:1650:14)
    at [eval]:1:24
    at Script.runInThisContext (vm.js:96:20)
    at Object.runInThisContext (vm.js:303:38)
    at Object.<anonymous> ([eval]-wrapper:6:22)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at evalScript (internal/bootstrap/node.js:582:27)
    at startup (internal/bootstrap/node.js:267:9)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:739:3)
```

Reply:
Hmm, I suppose that'd be an argument against the refactor, since realpath has 2 calls that take callbacks and one that doesn't.

There might be some benefit to also including a node-compatible API version of realpath that takes the path and callback args.

Reply:
ditched the refactor attempt, see the next top-level comment in this PR for an update